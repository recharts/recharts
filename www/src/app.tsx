/* eslint-disable import/no-unresolved */
import { createRoot } from 'react-dom/client';
import { loader } from '@monaco-editor/react';
import * as monaco from 'monaco-editor';
// @ts-ignore
import EditorWorker from 'monaco-editor/esm/vs/editor/editor.worker?worker';
// @ts-ignore
import JsonWorker from 'monaco-editor/esm/vs/language/json/json.worker?worker';
// @ts-ignore
import CssWorker from 'monaco-editor/esm/vs/language/css/css.worker?worker';
// @ts-ignore
import HtmlWorker from 'monaco-editor/esm/vs/language/html/html.worker?worker';
// @ts-ignore
import TsWorker from 'monaco-editor/esm/vs/language/typescript/ts.worker?worker';
import Root from './containers/Root';

// eslint-disable-next-line no-restricted-globals
self.MonacoEnvironment = {
  getWorker(_, label) {
    if (label === 'json') {
      return new JsonWorker();
    }
    if (label === 'css' || label === 'scss' || label === 'less') {
      return new CssWorker();
    }
    if (label === 'html' || label === 'handlebars' || label === 'razor') {
      return new HtmlWorker();
    }
    if (label === 'typescript' || label === 'javascript') {
      return new TsWorker();
    }
    return new EditorWorker();
  },
};

loader.config({ monaco });

loader.init().then(/* ... */);

const container = document.getElementById('app');
if (container == null) {
  throw new Error('Container element with id "app" not found');
}
const root = createRoot(container);
root.render(<Root />);
