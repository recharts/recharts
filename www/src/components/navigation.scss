@use '../styles/variables' as variables;

@media (max-width: variables.$mobile-breakpoint) {
  nav.mobile-only {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    flex: 1;
  }
}

.mobile-nav-toggle {
  position: absolute;
  cursor: pointer;
  background: none;
  border: none;
  padding: 0 20px;
  outline: none;
  z-index: 1001;
  display: flex;
  align-items: center;
  gap: 8px;

  &:focus {
    outline: 2px solid variables.$brand-info;
  }

  .hamburger-icon span {
    display: block;
    width: 24px;
    height: 3px;
    background-color: variables.$brand-info;
    margin-bottom: 5px;
    transition: transform 0.3s, opacity 0.3s;

    &:last-child {
      margin-bottom: 0;
    }
  }
}

.mobile-nav-backdrop {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  z-index: 999;
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.3s, visibility 0.3s;

  &.open {
    opacity: 1;
    visibility: visible;
  }
}

@media (min-width: variables.$mobile-breakpoint) {
  nav.desktop-only {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    gap: 40px;
    flex: 1;

    .navigation-items {
      display: flex;
      flex-direction: row;
      flex: 1;
      gap: 40px;

      a {
        display: inline-block;
        height: 61px;
        font-size: variables.$font-size-lg;

        &.active {
          color: variables.$brand-primary;
          border-bottom: 2px solid variables.$brand-primary;
          cursor: default;
        }
      }
    }
  }
}

.slide-right-mobile-nav {
  position: fixed;
  top: 0;
  right: 0;
  width: 80%;
  max-width: 500px;
  height: 100%;
  background: #fff;
  box-shadow: -2px 0 5px rgba(0, 0, 0, 0.1);
  transform: translateX(100%);
  transition: transform 0.3s ease-in-out;
  z-index: 1000;
  padding: 60px 0 20px;
  overflow-y: auto;

  &.open {
    transform: translateX(0);
  }

  .navigation-items {
    display: flex;
    flex-direction: column;
    gap: 15px;
    padding: 0;
    margin: 0;
    list-style: none;
    line-height: 2em;

    a {
      display: block;
      padding: 8px 20px;
      font-size: variables.$font-size-lg;

      &.active {
        color: variables.$brand-primary;
        text-decoration: underline;
      }
    }

    .collapsible {
      .collapsible-trigger {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 100%;
        background: transparent;
        border: none;
        color: inherit;
        font-family: inherit;
        font-size: variables.$font-size-lg;
        padding: 8px 20px;
        cursor: pointer;
        text-align: left;

        i {
          transition: transform 0.2s ease-in-out;
        }
      }

      &.expanded .collapsible-trigger i {
        transform: rotate(-90deg);
      }

      .collapsible-content .sidebar-cate {
        a {
          display: block;
          padding: 8px 20px;
        }

        h4 {
          margin: 0;
          font-size: variables.$font-size-lg;
        }
      }
    }
  }
}

body.no-scroll {
  overflow: hidden;
}
