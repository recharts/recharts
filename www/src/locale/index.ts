import zhCN from './zh-CN';
import enUS from './en-US';

export type SupportedLocale = 'zh-CN' | 'en-US';

const localeMap: Record<SupportedLocale, unknown> = {
  'zh-CN': zhCN,
  'en-US': enUS,
};

export const supportedLocales: ReadonlyArray<SupportedLocale> = ['zh-CN', 'en-US'];

export default localeMap;
