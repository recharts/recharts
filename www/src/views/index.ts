import IndexView from './IndexView';
import GuideView from './GuideView';
import APIView from './APIView';
import ExamplesView from './ExamplesView';
import Storybook from './Storybook';

export { IndexView, GuideView, APIView, ExamplesView, Storybook };
