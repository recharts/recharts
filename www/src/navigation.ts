import { ReactNode } from 'react';
import { SupportedLocale } from './locale';
import { localeGet, useLocale } from './utils/LocaleUtils.ts';
import { apiCates } from './docs/apiCates.ts';
import { allExamples } from './docs/exampleComponents';
import { allGuides } from './views/GuideView.tsx';

/**
 * Represents a single navigation link in the navigation menu.
 */
export type NavItem = {
  /**
   * The unique identifier for the item. Used as identified and i18n ID.
   */
  key: string;
  /**
   * The display name for the navigation link.
   * This can be a string or a React component for more complex rendering.
   */
  displayName: ReactNode;
  /**
   * The URL path for the navigation link.
   */
  url: string;
  /**
   * Optional children navigation items, allowing for nested menus.
   * This can be used to create sub-menus or hierarchical navigation structures.
   */
  categories?: ReadonlyArray<NavCategory>;
  /**
   * If true, the link will open in a new browser tab or window.
   * Defaults to false.
   */
  targetBlank?: boolean;
};

export type NavCategory = {
  /**
   * The unique identifier for the category. Used as identifier and i18n ID.
   */
  key: string;
  /**
   * The name of the category. Can be an empty string for uncategorized items.
   */
  displayName: string;
  /**
   * The list of navigation items under this category.
   */
  items: ReadonlyArray<NavItem>;
};

export type Navigation = ReadonlyArray<NavItem>;

function guideNavItems(locale: SupportedLocale): ReadonlyArray<NavCategory> {
  const items = allGuides.map(name => ({
    key: name,
    displayName: localeGet(locale, 'guide', name),
    url: `/${locale}/guide/${name}`,
  }));
  // The "Guide" page has only one category, without any names. As we add more categories in the future, we will adjust this.
  return [
    {
      key: 'guide',
      displayName: '',
      items,
    },
  ];
}

function apiNavItems(locale: SupportedLocale): ReadonlyArray<NavCategory> {
  return apiCates.map(({ name, items }) => ({
    key: name,
    displayName: localeGet(locale, 'api', name),
    items: items.map(compName => ({
      key: compName,
      displayName: compName,
      url: `/${locale}/api/${compName}`,
    })),
  }));
}

const exampleCategoryNames = Object.keys(allExamples).sort((a, b) => allExamples[a].order - allExamples[b].order);
function exampleNavItems(locale: SupportedLocale): ReadonlyArray<NavCategory> {
  return exampleCategoryNames.map(
    (cate): NavCategory => ({
      key: cate,
      displayName: cate,
      items: Object.keys(allExamples[cate].examples).map(name => ({
        key: name,
        displayName: allExamples[cate].examples[name].name,
        url: `/${locale}/examples/${name}`,
      })),
    }),
  );
}

export function getAllNavigationItems(locale: SupportedLocale): Navigation {
  return [
    {
      key: 'guide',
      displayName: localeGet(locale, 'frame', 'guide'),
      url: `/${locale}/guide`,
      categories: guideNavItems(locale),
    },
    {
      key: 'api',
      displayName: localeGet(locale, 'frame', 'api'),
      url: `/${locale}/api`,
      categories: apiNavItems(locale),
    },
    {
      key: 'examples',
      displayName: localeGet(locale, 'frame', 'examples'),
      url: `/${locale}/examples`,
      categories: exampleNavItems(locale),
    },
    { key: 'storybook', displayName: localeGet(locale, 'frame', 'storybook'), url: `/${locale}/storybook` },
    { key: 'wiki', displayName: 'Wiki', url: 'https://github.com/recharts/recharts/wiki', targetBlank: true },
    { key: 'github', displayName: 'GitHub', url: 'https://github.com/recharts/recharts', targetBlank: true },
  ];
}

export function useAllNavigationItems(): Navigation {
  const locale = useLocale();
  return getAllNavigationItems(locale);
}
