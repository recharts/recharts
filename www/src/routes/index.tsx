import { Navigate, Route, Routes, useParams } from 'react-router';
import { IndexView, GuideView, APIView, ExamplesView, Storybook } from '../views';
import Frame from '../layouts/Frame';
import { defaultLocale } from '../utils/LocaleUtils.ts';

function RedirectToDefaultExample() {
  const { locale } = useParams();
  return <Navigate to={`/${locale}/examples/SimpleLineChart`} replace />;
}

export default function routes() {
  return (
    <Routes>
      <Route path="/" element={<Navigate to={`/${defaultLocale}`} />} />
      <Route
        path="/:locale/guide/:name?"
        element={
          <Frame>
            <GuideView />
          </Frame>
        }
      />
      <Route
        path="/:locale/api/:name?"
        element={
          <Frame>
            <APIView />
          </Frame>
        }
      />
      <Route path="/:locale/examples" element={<RedirectToDefaultExample />} />
      <Route
        path="/:locale/examples/:name?"
        element={
          <Frame>
            <ExamplesView />
          </Frame>
        }
      />
      <Route
        path="/:locale/storybook"
        element={
          <Frame>
            <Storybook />
          </Frame>
        }
      />
      <Route
        path="*"
        element={
          <Frame>
            <IndexView />
          </Frame>
        }
      />
    </Routes>
  );
}
