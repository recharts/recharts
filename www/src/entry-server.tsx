import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router';
import { routes } from './routes';
import { supportedLocales } from './locale';
import { getSiteRoutes } from './navigation.data';

export function render(url: string, template: string) {
  const appHtml = renderToString(<StaticRouter location={url}>{routes()}</StaticRouter>);

  return template.replace('<div id="app"></div>', `<div id="app">${appHtml}</div>`);
}

/*
 * Your IDE may flag this function as unused, but actually it is used by prerender.tsx.
 * It's imported dynamically from the autogenerated file so your IDE may not be able to discover that.
 */
// noinspection JSUnusedGlobalSymbols
export function getAllRoutes(): string[] {
  const allRoutes: string[] = ['/'];

  // Get base routes from getSiteRoutes (without locale prefix)
  const baseRoutes = getSiteRoutes();
  allRoutes.push(...baseRoutes);

  // Add routes with locale prefixes for all locales
  supportedLocales.forEach(locale => {
    allRoutes.push(`/${locale}`);
    baseRoutes.forEach(route => {
      allRoutes.push(`/${locale}${route}`);
    });
  });

  return allRoutes;
}
