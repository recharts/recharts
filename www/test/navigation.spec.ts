import { describe, it, expect } from 'vitest';
import { getAllNavigationItems } from '../src/navigation';

describe('getAllNavigationItems', () => {
  it('should return all navigation items', () => {
    const locale = 'en-US';
    const navigationItems = getAllNavigationItems(locale);
    expect(navigationItems).toMatchSnapshot();
  });
});
