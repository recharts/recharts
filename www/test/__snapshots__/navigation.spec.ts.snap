// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAllNavigationItems > should return all navigation items 1`] = `
[
  {
    "categories": [
      {
        "NavPreview": null,
        "displayName": "",
        "items": [
          {
            "displayName": "Installation",
            "key": "installation",
            "url": "/en-US/guide/installation/",
          },
          {
            "displayName": "Getting Started",
            "key": "getting-started",
            "url": "/en-US/guide/getting-started/",
          },
          {
            "displayName": "Customize",
            "key": "customize",
            "url": "/en-US/guide/customize/",
          },
          {
            "displayName": "Active Index",
            "key": "activeIndex",
            "url": "/en-US/guide/activeIndex/",
          },
          {
            "displayName": "Chart size",
            "key": "sizes",
            "url": "/en-US/guide/sizes/",
          },
        ],
        "key": "guide",
      },
    ],
    "displayName": "Guide",
    "key": "guide",
    "url": "/en-US/guide/",
  },
  {
    "categories": [
      {
        "NavPreview": null,
        "displayName": "Charts",
        "items": [
          {
            "displayName": "AreaChart",
            "key": "AreaChart",
            "url": "/en-US/api/AreaChart/",
          },
          {
            "displayName": "BarChart",
            "key": "BarChart",
            "url": "/en-US/api/BarChart/",
          },
          {
            "displayName": "LineChart",
            "key": "LineChart",
            "url": "/en-US/api/LineChart/",
          },
          {
            "displayName": "ComposedChart",
            "key": "ComposedChart",
            "url": "/en-US/api/ComposedChart/",
          },
          {
            "displayName": "PieChart",
            "key": "PieChart",
            "url": "/en-US/api/PieChart/",
          },
          {
            "displayName": "RadarChart",
            "key": "RadarChart",
            "url": "/en-US/api/RadarChart/",
          },
          {
            "displayName": "RadialBarChart",
            "key": "RadialBarChart",
            "url": "/en-US/api/RadialBarChart/",
          },
          {
            "displayName": "ScatterChart",
            "key": "ScatterChart",
            "url": "/en-US/api/ScatterChart/",
          },
          {
            "displayName": "FunnelChart",
            "key": "FunnelChart",
            "url": "/en-US/api/FunnelChart/",
          },
          {
            "displayName": "Treemap",
            "key": "Treemap",
            "url": "/en-US/api/Treemap/",
          },
          {
            "displayName": "SankeyChart",
            "key": "SankeyChart",
            "url": "/en-US/api/SankeyChart/",
          },
        ],
        "key": "charts",
      },
      {
        "NavPreview": null,
        "displayName": "General Components",
        "items": [
          {
            "displayName": "ResponsiveContainer",
            "key": "ResponsiveContainer",
            "url": "/en-US/api/ResponsiveContainer/",
          },
          {
            "displayName": "Legend",
            "key": "Legend",
            "url": "/en-US/api/Legend/",
          },
          {
            "displayName": "Tooltip",
            "key": "Tooltip",
            "url": "/en-US/api/Tooltip/",
          },
          {
            "displayName": "Cell",
            "key": "Cell",
            "url": "/en-US/api/Cell/",
          },
          {
            "displayName": "Text",
            "key": "Text",
            "url": "/en-US/api/Text/",
          },
          {
            "displayName": "Label",
            "key": "Label",
            "url": "/en-US/api/Label/",
          },
          {
            "displayName": "LabelList",
            "key": "LabelList",
            "url": "/en-US/api/LabelList/",
          },
          {
            "displayName": "Customized",
            "key": "Customized",
            "url": "/en-US/api/Customized/",
          },
        ],
        "key": "general-components",
      },
      {
        "NavPreview": null,
        "displayName": "Cartesian Components",
        "items": [
          {
            "displayName": "Area",
            "key": "Area",
            "url": "/en-US/api/Area/",
          },
          {
            "displayName": "Bar",
            "key": "Bar",
            "url": "/en-US/api/Bar/",
          },
          {
            "displayName": "Line",
            "key": "Line",
            "url": "/en-US/api/Line/",
          },
          {
            "displayName": "Scatter",
            "key": "Scatter",
            "url": "/en-US/api/Scatter/",
          },
          {
            "displayName": "XAxis",
            "key": "XAxis",
            "url": "/en-US/api/XAxis/",
          },
          {
            "displayName": "YAxis",
            "key": "YAxis",
            "url": "/en-US/api/YAxis/",
          },
          {
            "displayName": "ZAxis",
            "key": "ZAxis",
            "url": "/en-US/api/ZAxis/",
          },
          {
            "displayName": "Brush",
            "key": "Brush",
            "url": "/en-US/api/Brush/",
          },
          {
            "displayName": "CartesianAxis",
            "key": "CartesianAxis",
            "url": "/en-US/api/CartesianAxis/",
          },
          {
            "displayName": "CartesianGrid",
            "key": "CartesianGrid",
            "url": "/en-US/api/CartesianGrid/",
          },
          {
            "displayName": "ReferenceLine",
            "key": "ReferenceLine",
            "url": "/en-US/api/ReferenceLine/",
          },
          {
            "displayName": "ReferenceDot",
            "key": "ReferenceDot",
            "url": "/en-US/api/ReferenceDot/",
          },
          {
            "displayName": "ReferenceArea",
            "key": "ReferenceArea",
            "url": "/en-US/api/ReferenceArea/",
          },
          {
            "displayName": "ErrorBar",
            "key": "ErrorBar",
            "url": "/en-US/api/ErrorBar/",
          },
          {
            "displayName": "Funnel",
            "key": "Funnel",
            "url": "/en-US/api/Funnel/",
          },
        ],
        "key": "cartesian-components",
      },
      {
        "NavPreview": null,
        "displayName": "Polar Components",
        "items": [
          {
            "displayName": "Pie",
            "key": "Pie",
            "url": "/en-US/api/Pie/",
          },
          {
            "displayName": "Radar",
            "key": "Radar",
            "url": "/en-US/api/Radar/",
          },
          {
            "displayName": "RadialBar",
            "key": "RadialBar",
            "url": "/en-US/api/RadialBar/",
          },
          {
            "displayName": "PolarAngleAxis",
            "key": "PolarAngleAxis",
            "url": "/en-US/api/PolarAngleAxis/",
          },
          {
            "displayName": "PolarGrid",
            "key": "PolarGrid",
            "url": "/en-US/api/PolarGrid/",
          },
          {
            "displayName": "PolarRadiusAxis",
            "key": "PolarRadiusAxis",
            "url": "/en-US/api/PolarRadiusAxis/",
          },
        ],
        "key": "polar-components",
      },
      {
        "NavPreview": null,
        "displayName": "Shapes",
        "items": [
          {
            "displayName": "Cross",
            "key": "Cross",
            "url": "/en-US/api/Cross/",
          },
          {
            "displayName": "Curve",
            "key": "Curve",
            "url": "/en-US/api/Curve/",
          },
          {
            "displayName": "Dot",
            "key": "Dot",
            "url": "/en-US/api/Dot/",
          },
          {
            "displayName": "Polygon",
            "key": "Polygon",
            "url": "/en-US/api/Polygon/",
          },
          {
            "displayName": "Rectangle",
            "key": "Rectangle",
            "url": "/en-US/api/Rectangle/",
          },
          {
            "displayName": "Sector",
            "key": "Sector",
            "url": "/en-US/api/Sector/",
          },
          {
            "displayName": "Trapezoid",
            "key": "Trapezoid",
            "url": "/en-US/api/Trapezoid/",
          },
        ],
        "key": "shapes",
      },
    ],
    "displayName": "API",
    "key": "api",
    "url": "/en-US/api/",
  },
  {
    "categories": [
      {
        "NavPreview": [Function],
        "displayName": "LineChart",
        "items": [
          {
            "displayName": "Simple Line Chart",
            "key": "SimpleLineChart",
            "url": "/en-US/examples/SimpleLineChart/",
          },
          {
            "displayName": "Tiny Line Chart",
            "key": "TinyLineChart",
            "url": "/en-US/examples/TinyLineChart/",
          },
          {
            "displayName": "Dashed Line Chart",
            "key": "DashedLineChart",
            "url": "/en-US/examples/DashedLineChart/",
          },
          {
            "displayName": "Vertical Line Chart",
            "key": "VerticalLineChart",
            "url": "/en-US/examples/VerticalLineChart/",
          },
          {
            "displayName": "Biaxial Line Chart",
            "key": "BiaxialLineChart",
            "url": "/en-US/examples/BiaxialLineChart/",
          },
          {
            "displayName": "Vertical Line Chart With Specified Domain",
            "key": "VerticalLineChartWithSpecifiedDomain",
            "url": "/en-US/examples/VerticalLineChartWithSpecifiedDomain/",
          },
          {
            "displayName": "Line Chart Connect Nulls",
            "key": "LineChartConnectNulls",
            "url": "/en-US/examples/LineChartConnectNulls/",
          },
          {
            "displayName": "Line Chart With X Axis Padding",
            "key": "LineChartWithXAxisPadding",
            "url": "/en-US/examples/LineChartWithXAxisPadding/",
          },
          {
            "displayName": "Line Chart With Reference Lines",
            "key": "LineChartWithReferenceLines",
            "url": "/en-US/examples/LineChartWithReferenceLines/",
          },
          {
            "displayName": "Customized Dot Line Chart",
            "key": "CustomizedDotLineChart",
            "url": "/en-US/examples/CustomizedDotLineChart/",
          },
          {
            "displayName": "Customized Label Line Chart",
            "key": "CustomizedLabelLineChart",
            "url": "/en-US/examples/CustomizedLabelLineChart/",
          },
          {
            "displayName": "Synchronized Line Chart",
            "key": "SynchronizedLineChart",
            "url": "/en-US/examples/SynchronizedLineChart/",
          },
          {
            "displayName": "Highlight And Zoom Line Chart",
            "key": "HighlightAndZoomLineChart",
            "url": "/en-US/examples/HighlightAndZoomLineChart/",
          },
          {
            "displayName": "Line Chart Has Multi Series",
            "key": "LineChartHasMultiSeries",
            "url": "/en-US/examples/LineChartHasMultiSeries/",
          },
          {
            "displayName": "Line Chart Axis Interval",
            "key": "LineChartAxisInterval",
            "url": "/en-US/examples/LineChartAxisInterval/",
          },
          {
            "displayName": "Line Chart Negative Values With Reference Lines",
            "key": "LineChartNegativeValuesWithReferenceLines",
            "url": "/en-US/examples/LineChartNegativeValuesWithReferenceLines/",
          },
        ],
        "key": "LineChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "AreaChart",
        "items": [
          {
            "displayName": "Simple Area Chart",
            "key": "SimpleAreaChart",
            "url": "/en-US/examples/SimpleAreaChart/",
          },
          {
            "displayName": "Stacked Area Chart",
            "key": "StackedAreaChart",
            "url": "/en-US/examples/StackedAreaChart/",
          },
          {
            "displayName": "Area Chart Connect Nulls",
            "key": "AreaChartConnectNulls",
            "url": "/en-US/examples/AreaChartConnectNulls/",
          },
          {
            "displayName": "Cardinal Area Chart",
            "key": "CardinalAreaChart",
            "url": "/en-US/examples/CardinalAreaChart/",
          },
          {
            "displayName": "Percent Area Chart",
            "key": "PercentAreaChart",
            "url": "/en-US/examples/PercentAreaChart/",
          },
          {
            "displayName": "Synchronized Area Chart",
            "key": "SynchronizedAreaChart",
            "url": "/en-US/examples/SynchronizedAreaChart/",
          },
          {
            "displayName": "Tiny Area Chart",
            "key": "TinyAreaChart",
            "url": "/en-US/examples/TinyAreaChart/",
          },
          {
            "displayName": "Area Chart Fill By Value",
            "key": "AreaChartFillByValue",
            "url": "/en-US/examples/AreaChartFillByValue/",
          },
        ],
        "key": "AreaChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "BarChart",
        "items": [
          {
            "displayName": "Tiny Bar Chart",
            "key": "TinyBarChart",
            "url": "/en-US/examples/TinyBarChart/",
          },
          {
            "displayName": "Simple Bar Chart",
            "key": "SimpleBarChart",
            "url": "/en-US/examples/SimpleBarChart/",
          },
          {
            "displayName": "Stacked Bar Chart",
            "key": "StackedBarChart",
            "url": "/en-US/examples/StackedBarChart/",
          },
          {
            "displayName": "Mix Bar Chart",
            "key": "MixBarChart",
            "url": "/en-US/examples/MixBarChart/",
          },
          {
            "displayName": "Custom Shape Bar Chart",
            "key": "CustomShapeBarChart",
            "url": "/en-US/examples/CustomShapeBarChart/",
          },
          {
            "displayName": "Positive and Negative Bar Chart",
            "key": "PositiveAndNegativeBarChart",
            "url": "/en-US/examples/PositiveAndNegativeBarChart/",
          },
          {
            "displayName": "Brush Bar Chart",
            "key": "BrushBarChart",
            "url": "/en-US/examples/BrushBarChart/",
          },
          {
            "displayName": "Bar Chart With Customized Event",
            "key": "BarChartWithCustomizedEvent",
            "url": "/en-US/examples/BarChartWithCustomizedEvent/",
          },
          {
            "displayName": "Bar Chart With Min Height",
            "key": "BarChartWithMinHeight",
            "url": "/en-US/examples/BarChartWithMinHeight/",
          },
          {
            "displayName": "Bar Chart Stacked By Sign",
            "key": "BarChartStackedBySign",
            "url": "/en-US/examples/BarChartStackedBySign/",
          },
          {
            "displayName": "Biaxial Bar Chart",
            "key": "BiaxialBarChart",
            "url": "/en-US/examples/BiaxialBarChart/",
          },
          {
            "displayName": "Bar Chart with background",
            "key": "BarChartHasBackground",
            "url": "/en-US/examples/BarChartHasBackground/",
          },
          {
            "displayName": "Bar Chart With Multi X Axis",
            "key": "BarChartWithMultiXAxis",
            "url": "/en-US/examples/BarChartWithMultiXAxis/",
          },
        ],
        "key": "BarChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "ComposedChart",
        "items": [
          {
            "displayName": "Line Bar Area Composed Chart",
            "key": "LineBarAreaComposedChart",
            "url": "/en-US/examples/LineBarAreaComposedChart/",
          },
          {
            "displayName": "Same Data Composed Chart",
            "key": "SameDataComposedChart",
            "url": "/en-US/examples/SameDataComposedChart/",
          },
          {
            "displayName": "Vertical Composed Chart",
            "key": "VerticalComposedChart",
            "url": "/en-US/examples/VerticalComposedChart/",
          },
          {
            "displayName": "Composed Chart With Axis Labels",
            "key": "ComposedChartWithAxisLabels",
            "url": "/en-US/examples/ComposedChartWithAxisLabels/",
          },
          {
            "displayName": "Scatter And Line Of Best Fit",
            "key": "ScatterAndLineOfBestFit",
            "url": "/en-US/examples/ScatterAndLineOfBestFit/",
          },
          {
            "displayName": "Banded Chart",
            "key": "BandedChart",
            "url": "/en-US/examples/BandedChart/",
          },
        ],
        "key": "ComposedChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "ScatterChart",
        "items": [
          {
            "displayName": "Simple Scatter Chart",
            "key": "SimpleScatterChart",
            "url": "/en-US/examples/SimpleScatterChart/",
          },
          {
            "displayName": "Three Dim Scatter Chart",
            "key": "ThreeDimScatterChart",
            "url": "/en-US/examples/ThreeDimScatterChart/",
          },
          {
            "displayName": "Joint Line Scatter Chart",
            "key": "JointLineScatterChart",
            "url": "/en-US/examples/JointLineScatterChart/",
          },
          {
            "displayName": "Bubble Chart",
            "key": "BubbleChart",
            "url": "/en-US/examples/BubbleChart/",
          },
          {
            "displayName": "Scatter Chart With Labels",
            "key": "ScatterChartWithLabels",
            "url": "/en-US/examples/ScatterChartWithLabels/",
          },
          {
            "displayName": "Multiple Y Axes Scatter Chart",
            "key": "MultipleYAxesScatterChart",
            "url": "/en-US/examples/MultipleYAxesScatterChart/",
          },
          {
            "displayName": "Scatter Chart With Cells",
            "key": "ScatterChartWithCells",
            "url": "/en-US/examples/ScatterChartWithCells/",
          },
          {
            "displayName": "Scatter Chart with many points (performance test)",
            "key": "ScatterChartPerformance",
            "url": "/en-US/examples/ScatterChartPerformance/",
          },
        ],
        "key": "ScatterChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "PieChart",
        "items": [
          {
            "displayName": "Two Level Pie Chart",
            "key": "TwoLevelPieChart",
            "url": "/en-US/examples/TwoLevelPieChart/",
          },
          {
            "displayName": "Straight Angle Pie Chart",
            "key": "StraightAnglePieChart",
            "url": "/en-US/examples/StraightAnglePieChart/",
          },
          {
            "displayName": "Custom Active Shape Pie Chart",
            "key": "CustomActiveShapePieChart",
            "url": "/en-US/examples/CustomActiveShapePieChart/",
          },
          {
            "displayName": "Pie Chart With Customized Label",
            "key": "PieChartWithCustomizedLabel",
            "url": "/en-US/examples/PieChartWithCustomizedLabel/",
          },
          {
            "displayName": "Pie Chart with gap and rounded corners",
            "key": "PieChartWithPaddingAngle",
            "url": "/en-US/examples/PieChartWithPaddingAngle/",
          },
          {
            "displayName": "Pie Chart With Needle",
            "key": "PieChartWithNeedle",
            "url": "/en-US/examples/PieChartWithNeedle/",
          },
          {
            "displayName": "Pie Chart in Flexbox",
            "key": "PieChartInFlexbox",
            "url": "/en-US/examples/PieChartInFlexbox/",
          },
          {
            "displayName": "Pie Chart in Grid",
            "key": "PieChartInGrid",
            "url": "/en-US/examples/PieChartInGrid/",
          },
        ],
        "key": "PieChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "RadarChart",
        "items": [
          {
            "displayName": "Simple Radar Chart",
            "key": "SimpleRadarChart",
            "url": "/en-US/examples/SimpleRadarChart/",
          },
          {
            "displayName": "Specified Domain Radar Chart",
            "key": "SpecifiedDomainRadarChart",
            "url": "/en-US/examples/SpecifiedDomainRadarChart/",
          },
        ],
        "key": "RadarChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "RadialBarChart",
        "items": [
          {
            "displayName": "Simple Radial Bar Chart",
            "key": "SimpleRadialBarChart",
            "url": "/en-US/examples/SimpleRadialBarChart/",
          },
        ],
        "key": "RadialBarChart",
      },
      {
        "NavPreview": [Function],
        "displayName": "TreeMap",
        "items": [
          {
            "displayName": "Simple Treemap",
            "key": "SimpleTreemap",
            "url": "/en-US/examples/SimpleTreemap/",
          },
          {
            "displayName": "Custom Content Treemap",
            "key": "CustomContentTreemap",
            "url": "/en-US/examples/CustomContentTreemap/",
          },
        ],
        "key": "TreeMap",
      },
      {
        "NavPreview": null,
        "displayName": "Tooltip",
        "items": [
          {
            "displayName": "Custom Content Of Tooltip",
            "key": "CustomContentOfTooltip",
            "url": "/en-US/examples/CustomContentOfTooltip/",
          },
        ],
        "key": "Tooltip",
      },
      {
        "NavPreview": null,
        "displayName": "Legend",
        "items": [
          {
            "displayName": "Legend with Opacity",
            "key": "LegendEffectOpacity",
            "url": "/en-US/examples/LegendEffectOpacity/",
          },
        ],
        "key": "Legend",
      },
      {
        "NavPreview": null,
        "displayName": "ResponsiveContainer",
        "items": [
          {
            "displayName": "Area Responsive Container",
            "key": "AreaResponsiveContainer",
            "url": "/en-US/examples/AreaResponsiveContainer/",
          },
          {
            "displayName": "Composed Responsive Container",
            "key": "ComposedResponsiveContainer",
            "url": "/en-US/examples/ComposedResponsiveContainer/",
          },
          {
            "displayName": "Pie Responsive Container",
            "key": "PieResponsiveContainer",
            "url": "/en-US/examples/PieResponsiveContainer/",
          },
        ],
        "key": "ResponsiveContainer",
      },
    ],
    "displayName": "Examples",
    "key": "examples",
    "url": "/en-US/examples/",
  },
  {
    "displayName": "Storybook",
    "key": "storybook",
    "url": "/en-US/storybook/",
  },
  {
    "displayName": "Wiki",
    "key": "wiki",
    "targetBlank": true,
    "url": "https://github.com/recharts/recharts/wiki",
  },
  {
    "displayName": "GitHub",
    "key": "github",
    "targetBlank": true,
    "url": "https://github.com/recharts/recharts",
  },
]
`;
